frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlertsRulesDetailsPage.tsx

```diff
@@ -40,24 +54,57 @@ const MonitoringAlertsDetailsPage: React.FC<MonitoringAlertsDetailsPageProps> =
   const namespace = match.params.ns;
   const { path } = match;
   const dispatch = useDispatch();
-  const [response, loadError, loading] = usePrometheusRulesPoll({ namespace });
-  const thanosAlertsAndRules = React.useMemo(
-    () => (!loading && !loadError ? getAlertsAndRules(response?.data) : { rules: [], alerts: [] }),
-    [response, loadError, loading],
+  const alerts = useSelector(({ observe }: RootState) => observe.get('devAlerts'));
+  const [customExtensions] = useResolvedExtensions<AlertingRulesSourceExtension>(
+    isAlertingRulesSource,
+  );
+  const alertsSource = React.useMemo(
+    () =>
+      customExtensions
+        .filter((extension) => extension.properties.contextId === 'dev-observe-alerting')
+        .map((extension) => extension.properties),
+    [customExtensions],
   );
 
   React.useEffect(() => {
-    const sortThanosRules = _.sortBy(thanosAlertsAndRules.rules, alertingRuleStateOrder);
-    dispatch(alertingSetRules('devRules', sortThanosRules, 'dev'));
-    dispatch(alertingLoaded('devAlerts', thanosAlertsAndRules.alerts, 'dev'));
-  }, [dispatch, thanosAlertsAndRules]);
+    const url = getPrometheusURL({
+      endpoint: PrometheusEndpoint.RULES,
+      namespace,
+    });
+    dispatch(alertingLoading('alerts', 'dev'));
+
+    const poller = (): void => {
```

Comment:
I notice that this code is almost the same as the `poller` we have in `/components/monitoring/alerting.tsx`. I think it would be worth extracting this into a single helper function.